openadapt/ml_models/huggingface_models/create_modal_image.py

```diff
@@ -0,0 +1,71 @@
+import modal
+
+GISTING_CONDA_IMAGE = (
+    modal.Image.conda()
+    .conda_install(
+        "cudatoolkit=11.2",
+        "cudnn=8.1.0",
+        "cuda-nvcc",
+        channels=["conda-forge", "nvidia"],
+    )
+    .apt_install("git")
+    .pip_install(
+        "git+https://github.com/huggingface/transformers@fb366b9a2a",
+        "gisting_test==0.2.3",
+        "accelerate==0.18.0",
+        "datasets==2.10.0",
+        "deepspeed==0.8.3",
+        "evaluate==0.3.0",
+        "fire==0.5.0",
+        "hydra-core==1.2.0",
+        "numpy==1.21.2",
+        "omegaconf>=2.1.1",
+        "openai==0.27.2",
+        "rouge_score==0.1.2",
+        "nltk==3.6.2",
+        "sentencepiece==0.1.98",
+        "torch==2.0.0",
+        "transformers",
+        "wandb==0.13.4",
+    )
+)
+
+BASE_PACKAGE_LIST = [
+    "accelerate==0.18.0",
+    "fire==0.5.0",
+    "hydra-core==1.2.0",
+    "numpy==1.21.2",
+    "openai==0.27.2",
+    "nltk==3.6.2",
+    "sentencepiece==0.1.98",
+    "torch==2.0.0",
+    "transformers",
+    "wandb==0.13.4",
+]
+
+
+def create_conda_image(package: str = None, packages: list[str] = None):
```

Comment:
I was working on a GistingCompletionProvider and thought that a file like this could be a useful addition to the model API, where we can specify what packages we would like the image to contain before running inference on a modal cloud instance. 

It is worth noting that modal purely hosts functions on the cloud, and not classes.